File: javabot-play/app/utils/Context.scala
package utils

import javabot.dao.ChannelDao
import javabot.model.Logs
import javax.inject.Inject
import java.time.LocalDateTime
import play.api.mvc
import play.api.mvc.AnyContent

class Context @Inject()(factoidDao: FactoidDao, channelDao: ChannelDao, adminDao: AdminDao, logsDao: LogsDao) {

  var logs: List[Logs] = null

  private var request: mvc.Request[AnyContent] = null

  def request(request: mvc.Request[AnyContent]) {
    this.request = request
  }

  var today = new DateTime

  var channel: String = null

  def factoidCount = factoidDao.count()

  def showAll: Boolean = {
    val userName = request.session.get("userName")

    userName.exists(name => {
      val admin = adminDao.getAdmin(name)
      admin != null && admin.getBotOwner
    })
  }

  def channels = {
    channelDao.findLogged(showAll)
  }

  def getLogs = {
    logs
  }

  def logChannel(name: String, date: DateTime) {
    channel = name
    logs = logsDao.findByChannel(channel, date, showAll)
  }
}

File: web/app/utils/Context.java
package utils;

import java.util.List;
import javax.inject.Inject;

import javabot.dao.ChannelDao;
import javabot.dao.FactoidDao;
import javabot.model.Admin;
import javabot.model.Logs;
import org.joda.time.DateTime;
import play.mvc.Http;

public class Context {
  @Inject
  private FactoidDao factoidDao;

  @Inject
  private ChannelDao channelDao;

  @Inject
  private AdminDao adminDao;

  @Inject
  private LogsDao logsDao;

  public List<Logs> logs;

  public String channel;

  public Long factoidCount() {
    return factoidDao.count();
  }

  public Boolean showAll() {
    String userName = Http.Context.current().session().get("userName");
    if (userName != null) {
      Admin admin = adminDao.getAdmin(userName);
      return admin != null && admin.getBotOwner();
    }
    return false;
  }

  public List<javabot.model.Channel> channels() {
    return channelDao.findLogged(showAll());
  }

  public List<Logs> getLogs() {
    return logs;
  }

  public void logChannel(String name, DateTime date) {
    channel = name;
    logs = logsDao.findByChannel(channel, date, showAll());
  }
}

